rdb_file_writer.go

```diff
@@ -0,0 +1,147 @@
+package rdb
+
+import (
+	"encoding/binary"
+	"io"
+	"os"
+)
+
+type FileWriter struct {
+	file *os.File
```

Comment:
we don't use any buffered io here, and making lots of small writes. i think we should use buffered io if possible, otherwise the rdb save can take quite a while

Reply:
I switched to buffered I/O. The position seek is a bit tricky, we have to flush all content before seeking. I also added a test to verify the position change with the buffered version.